src/v/config/property.h

```diff
@@ -953,5 +963,115 @@ class hidden_when_default_property : public property<T> {
         return this->value() == this->default_value();
     }
 };
+namespace detail {
+
+template<typename P>
+concept Property = requires() {
+    std::derived_from<P, base_property>;
+    typename P::underlying;
+};
+
+template<typename T>
+concept Array = !(std::is_same_v<std::decay_t<T>, ss::sstring>
+                  || std::is_same_v<std::decay_t<T>, std::string>)
+                && is_collection<T>;
```

Comment:
Might be worth unifying this with `detail::is_array` in this file.

Reply:
yeah i have a note to revisit this. iirc they're slightly different so I sidestepped it here.